components/TaskCard.vue

```diff
@@ -5,32 +5,51 @@
         <LoadingIcon />
       </div>
     </div>
-    <div class="card">
-      <div class="text-description" :class="{ 'text-done': task.done }">
-        {{ task.description }}
-      </div>
-      <div class="text-caption margin-top-9 margin-bottom-24">
-        {{ formatDate }}
-      </div>
-      <div class="space-between flex-box">
-        <div class="flex-gap-8 card-button">
-          <div v-if="!task.done" class="flex-gap-8">
-            <button @click="markDone">
-              <img src="@/assets/svg/Tick.svg" />
-            </button>
-            <button>
-              <img src="@/assets/svg/Edit.svg" />
+    <form @submit.prevent="checkForm">
+      <div class="card">
+        <div
+          v-if="showEditIcon"
+          class="text-description"
+          :class="{ 'text-done': task.done }"
+        >
+          {{ task.description }}
+        </div>
+        <div v-else>
+          <textarea id="title" v-model="taskDescription"></textarea>
+          <label v-if="titleInputError" for="title">{{
+            $t('validation.todo.title.required')
+          }}</label>
+        </div>
+        <div class="text-caption margin-top-9 margin-bottom-24">
+          {{ formatDate }}
+        </div>
+        <div class="space-between flex-box">
+          <div class="flex-gap-8 card-button">
+            <div v-if="!task.done" class="flex-gap-8">
+              <button value="update" @click.prevent="markDone">
+                <img src="@/assets/svg/Tick.svg" />
+              </button>
+              <button
+                v-if="showEditIcon"
+                value="edit"
+                @click.prevent="showEditIcon = false"
+              >
+                <img src="@/assets/svg/Edit.svg" />
+              </button>
+              <button v-else type="submit">
+                {{ $t('Save') }}
+              </button>
+            </div>
+            <button @click.prevent="deleteTask">
+              <img src="@/assets/svg/Delete.svg" />
```

Comment:
do not use SVGs as images.

Reply:
I fixed this in pagination branch. should i fix it here?